### ISSUE TITLE
Google BigQuery External Table

### ISSUE THREAD

**Issue by AHZenan** [apache/superset#18880]
#### How to reproduce the bug

1. Add a Google Big Query Database that has at least 2 native tables within 
2. Attempt to add the datasets from the database to the superset platform
3. You should be able to add the datasets easily.
4. Now create an external table from gsheets on the original Big Query database . Refresh superset and repeat step 2. 

### Expected results

After repeating step 2 expected to see 3 datasets , 2 native and 1 new external table 

### Actual results

After repeating step 2 , I cannot see the newly added external table

### Environment

(please complete the following information):

- browser type and version:
- superset version: `superset version`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by srinify** [comment]
@AHZenan if you write / run a query in Superset SQL Lab, are you able to get back results?

**Comment by kzosabe** [comment]
I reproduced this problem in my environment.
The cause of the problem is that including external resources in schema is only implemented in `sqlalchemy-bigquery` version 1.2.1 or later.
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/main/CHANGELOG.md#121-2021-10-27

The current workaround is to remove `pybigquery` from your superset dependencies and include `sqlalchemy-bigquery` instead to get the external resource.

Since `pybigquery` is an old library that is no longer maintained, it would be a good idea for superset to officially support `sqlalchemy-bigquery`.

```
Attention!
pybigquery is obsolete. Use sqlalchemy-bigquery instead. All future changes will be made to sqlalchemy-bigquery.
```
https://pypi.org/project/pybigquery/

**Comment by AHZenan** [comment]
@srinify yes I tried writing it in the SQL lab , it did not recognize the link. I am going to try re installing now using the sqlalchemy-bigquery instead ! @kzosabe thanks!

**Comment by AHZenan** [comment]
@srinify I think superset only shows native tables in the list . The way to access it is via the SQL Lab . However ,we just cannot use the gsheet link or just use the name of the file . We need to add the project name like so :

SELECT* FROM projectname.schema.filename

Cheers!

**Comment by srinify** [comment]
@AHZenan yeah that's correct. Referring to the Google Sheets URL won't work but you'll need the project.schema.filename syntax as you suggested!

**Comment by AHZenan** [comment]
@kzosabe  am i just supposed to re install superset using help install and configure the yaml file with sqlalchemy-bigquery? I tried it by just replacing the pybigquery with what you suggested but its not taking it

**Comment by kzosabe** [comment]
@AHZenan I'm not sure if I'm understanding your situation correctly, but if you replace pybigquery with sqlalchemy-bigquery and follow the installation procedure again, you should be able to view the external tables.
For example, I can display the external table in the table schema area of SQL Lab by the following steps.
After cloning the latest superset, 

```
echo "sqlalchemy-bigquery" > ./docker/requirements-local.txt
docker-compose -f docker-compose-non-dev.yml pull
docker-compose -f docker-compose-non-dev.yml up
```

The procedure is as described in the following document, but may vary if your installation follows a different method.
https://superset.apache.org/docs/databases/docker-add-drivers

These steps are enough to display the items in the table schema area or to create dataset, but if you actually try to retrieve the contents, you may get permission denied error.
In that case, you can get the content by sharing the spreadsheet to the your service account email address which used by your Superset BigQuery connection.

**Comment by rusackas** [comment]
It _sounds like_ this thread went quiet because the solutions in the thread worked out, so I'll close it. If anyone sees a need to update documentation or further address a root cause, I'm more than happy to re-ope this issue.
